internal/models/oauth_authorization.go

```diff
@@ -0,0 +1,279 @@
+
+const (
+	OAuthServerResponseTypeCode OAuthServerResponseType = "code"
+)
+
+func (rt OAuthServerResponseType) String() string {
+	return string(rt)
+}
+
+// OAuthServerAuthorization represents an OAuth 2.1 server authorization request
+type OAuthServerAuthorization struct {
+	ID                  uuid.UUID                      `json:"-" db:"id"`
+	AuthorizationID     string                         `json:"authorization_id" db:"authorization_id"`
+	ClientID            string                         `json:"client_id" db:"client_id"`
+	UserID              *uuid.UUID                     `json:"user_id" db:"user_id"`
+	RedirectURI         string                         `json:"redirect_uri" db:"redirect_uri"`
+	Scope               string                         `json:"scope" db:"scope"`
+	State               storage.NullString             `json:"state" db:"state"`
+	CodeChallenge       storage.NullString             `json:"code_challenge" db:"code_challenge"`
+	CodeChallengeMethod storage.NullString             `json:"code_challenge_method" db:"code_challenge_method"`
+	ResponseType        OAuthServerResponseType        `json:"response_type" db:"response_type"`
+	Status              OAuthServerAuthorizationStatus `json:"status" db:"status"`
+	AuthorizationCode   storage.NullString             `json:"-" db:"authorization_code"`
+	CreatedAt           time.Time                      `json:"created_at" db:"created_at"`
+	ExpiresAt           time.Time                      `json:"expires_at" db:"expires_at"`
+	ApprovedAt          *time.Time                     `json:"approved_at" db:"approved_at"`
+
+	// Relations with OAuth clients
+	Client *OAuthServerClient `json:"client,omitempty" db:"-"`
+}
+
+// TableName returns the table name for the OAuthServerAuthorization model
+func (OAuthServerAuthorization) TableName() string {
+	return "oauth_authorizations"
+}
+
+// NewOAuthServerAuthorization creates a new OAuth server authorization request without user (for initial flow)
+func NewOAuthServerAuthorization(clientID, redirectURI, scope, state, codeChallenge, codeChallengeMethod string) *OAuthServerAuthorization {
+	id := uuid.Must(uuid.NewV4())
+	authorizationID := crypto.SecureAlphanumeric(32) // Generate random ID for frontend
+
+	now := time.Now()
+	expiresAt := now.Add(10 * time.Minute) // 10 minute expiration
+
+	auth := &OAuthServerAuthorization{
+		ID:              id,
+		AuthorizationID: authorizationID,
+		ClientID:        clientID,
+		UserID:          nil, // No user yet
+		RedirectURI:     redirectURI,
+		Scope:           scope,
+		ResponseType:    OAuthServerResponseTypeCode,
+		Status:          OAuthServerAuthorizationPending,
+		CreatedAt:       now,
+		ExpiresAt:       expiresAt,
+	}
+
+	if state != "" {
+		auth.State = storage.NullString(state)
+	}
+	if codeChallenge != "" {
+		auth.CodeChallenge = storage.NullString(codeChallenge)
+	}
+	if codeChallengeMethod != "" {
+		auth.CodeChallengeMethod = storage.NullString(codeChallengeMethod)
+	}
```

Comment:
Would it be too much trouble to instead use pointer strings? They're a bit clunky, but we've been burned at least 3 times with `NullString` so I'd rather we stop using it for anything new. The semantics aren't very clear with it especially when debugging subtle issues under pressure.